lib/extensions/src/lib/services/extension.service.ts

```diff
@@ -58,6 +59,8 @@ export class ExtensionService {
     features: Array<any> = [];
     authGuards: { [key: string]: Type<{}> } = {};
 
+    readonly setup$ = new BehaviorSubject<ExtensionConfig>(this.config);
```

Comment:
Can you please reveal the Observable interface only here? I don't think making it readonly helps us here, someone from outside can still call `next()` on this BehaviourSubject.

Reply:
how should a unit test interact with these events? replacing the entire service instance with a custom one?

Reply:
@popovicsandras can you please check the updated code

Reply:
A common technique for this is that for the sake of unit test, you can inject a very tiny factory function as the last parameter of the constructor:
```
function defaultSubjectFactory<T>(initParam) {
    return new BehaviorSubject<T>(initParam);
}

@Injectable({providedIn: 'root'})
export class ExtensionService {

    // ...

    private setupSubject$: BehaviorSubject<ExtensionConfig>;
    setup$: Observable<ExtensionConfig>;

    constructor(/* ...every other constructor parameter, */ subjectFactory = defaultSubjectFactory) {
        // ...
        this.setupSubject$ = subjectFactory<ExtensionConfig>(this.config);
        this.setup$ = this.setupSubject$.asObservable();
        // ...
    }

    // ...
}
```

Reply:
solved with protected member and custom service mock for unit tests